nuxeo-ai-core/src/main/java/org/nuxeo/ai/auto/AutoServiceImpl.java

```diff
@@ -157,11 +163,14 @@ protected void calculateAutoCorrect(SuggestionMetadataWrapper docMetadata, Strin
         }
     }
 
-    protected boolean setProperty(Property property, String newValue) {
-        Serializable currentValue = property.getValue();
+    protected boolean setProperty(CoreSession session, DocumentModel doc, String key, Serializable currentValue, String newValue) {
         if (!newValue.equals(currentValue)) {
-            property.setValue(newValue);
-            return true;
+            try {
+                DocumentHelper.setProperty(session, doc, key, newValue);
```

Comment:
a bit confused why do we need this helper when setting a property is very straightforward and the advantage of such is questionable

Reply:
I am using it because it supports multi-value fields.

Reply:
ok